exporter/otelarrowexporter/go.mod

```diff
@@ -0,0 +1,92 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/otelarrowexporter
+
+go 1.20
+
+require (
+	github.com/open-telemetry/otel-arrow v0.16.0
+	github.com/open-telemetry/otel-arrow/collector v0.16.0
```

Comment:
these dependencies will eventually go away once the implementation is complete in this repo correct?

Reply:
The `otel-arrow` module contains a large piece of code for translating between Arrow and `pdata` OTLP. This library probably has uses outside of the Collector and I think it should remain separate, at least until the Arrow proto definitions are accepted formally as OTLP protocol.  There are OTel-Go SDK reasons we might use that library outside of collector components, probably.

The `otel-arrow/collector` module contains a number of components that were useful for us during development of the primary exporter/receiver, and we will want to keep them. They could each move into the `-contrib` repository, but that will take some additional effort, so I expect the repository itself to remain.

In addition to those statements, there are three packages for which an answer is needed, these being shared by the exporter and receiver and things I would recommend _for the core collector itself_. 

The `otel-arrow/collector/sharedcomponent` package is a copy of the one from the core repo's internal/ package. I'm not sure why the core repo won't export that functionality, so I replicated it.

[The `otel-arrow/collector/netstats` package is a set of very useful metrics that the Arrow components use, but which are applicable to all components](https://github.com/open-telemetry/opentelemetry-collector/issues/6638) and would also  useful for the automatic benchmark harness.

The `otel-arrow/collector/compression` package is an implementation that supports multiple configurable Zstd compression levels on a per-exporter component basis. See https://github.com/mostynb/go-grpc-compression/issues/24.

I suspect you will agree that all three of these packages belong in either `-collector` or `-collector-contrib`.